index_settings.go

```diff
@@ -445,3 +445,52 @@ func (i Index) ResetSortableAttributes() (resp *Task, err error) {
 	}
 	return resp, nil
 }
+
+func (i Index) GetTypoTolerance() (resp *TypoTolerance, err error) {
+	resp = &TypoTolerance{}
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/settings/typo-tolerance",
+		method:              http.MethodGet,
+		withRequest:         nil,
+		withResponse:        resp,
+		acceptedStatusCodes: []int{http.StatusOK},
+		functionName:        "GetTypoTolerance",
+	}
+	if err := i.client.executeRequest(req); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func (i Index) UpdateTypoTolerance(request *TypoTolerance) (resp *Task, err error) {
+	resp = &Task{}
+	req := internalRequest{
+		endpoint:            "/indexes/" + i.UID + "/settings/typo-tolerance",
+		method:              http.MethodPost,
+		contentType:         contentTypeJSON,
+		withRequest:         &request,
```

Comment:
Can you enlighten me on this one?

Reply:
`Object` doesn't really exist in go, so to send multiple arguments it's easier to use struct. For this purpose, we create this struct with all data we need to process a request:
```go
type internalRequest struct {
	endpoint    string
	method      string
	contentType string

	withRequest     interface{}
	withResponse    interface{}
	withQueryParams map[string]string

	acceptedStatusCodes []int

	functionName string
}
```
We send the struct to the eexecuteRequest` function with all the data needed, and a `interface{}` in `withRequest` who will be fill with the response from the API.